src/benchmark.py

```diff
@@ -0,0 +1,235 @@
+import os
+import sys
+import time
+import datetime
+import pytz
+import boto
+import boto3
+import s3fs
+import pandas as pd
+import json
+import psycopg2
+import sqlalchemy as sa # Package for accessing SQL databases via Python
+import StringIO
+
+def pd_to_postgres(df, table_name, mode):
+    """
+    Save DataFrame to PostgreSQL by to_sql() function.
+    """
+    # Writing Dataframe to PostgreSQL and replacing table if it already exists
+    df.to_sql(name=table_name, con=engine, if_exists = mode, index=False)
+
+def cleanColumns(columns):
+    cols = []
+    for col in columns:
+        col = col.replace(' ', '_')
+        cols.append(col)
+    return cols
+
+def df_to_sql(df, table_name, mode, new_table, psql, cur, engine):
+    """
+    Save DataFrame to .csv, read csv as sql table in memory and copy the table
+     directly in batch to PostgreSQL or Redshift.
+    """
+    # Prepare schema for table
+    df[:0].to_sql('temp', engine, if_exists = "replace", index=False)
+    df.columns = cleanColumns(df.columns)
+    # replace mode
+    if mode == 'replace':
+        cur.execute("DROP TABLE " + table_name)
+        cur.connection.commit()
+    # Prepare data to temp
+    if psql == 'psql':
+        data = StringIO.StringIO()
+        df.to_csv(data, header=False, index=False)
+        data.seek(0)
+        sql_query = """
+            COPY temp FROM STDIN WITH
+                CSV
+                HEADER;
+        """
+        cur.copy_expert(sql=sql_query, file=data)
+        cur.connection.commit()
+        # Copy or append table temp to target table
+        if (mode == 'replace' or new_table == 0):
```

Comment:
same comment here. Instead of checking mode=='replace', making this an ENUM. Two possible modes are possible REPLACE OR APPEND